File: extract.js
const fs = require('fs');
const html = fs.readFileSync('scraped.html', 'utf-8');

// Simple regex to extract text
let text = html.replace(/<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi, ' ');
text = text.replace(/<style\b[^<]*(?:(?!<\/style>)<[^<]*)*<\/style>/gi, ' ');
text = text.replace(/<[^>]+>/g, ' ');
text = text.replace(/\s+/g, ' ').trim();

fs.writeFileSync('extracted.txt', text);

// Extract links
const linkRegex = /href="(https:\/\/skyomcity\.in\/[^"]*)"/g;
const links = new Set();
let match;
while ((match = linkRegex.exec(html)) !== null) {
    links.add(match[1]);
}
fs.writeFileSync('links.txt', Array.from(links).join('\n'));

console.log('Extraction complete.');

File: extract.py
import re

with open('scraped.html', 'r', encoding='utf-8') as f:
    html = f.read()

# Remove scripts and styles
html = re.sub(r'<script.*?>.*?</script>', ' ', html, flags=re.DOTALL)
html = re.sub(r'<style.*?>.*?</style>', ' ', html, flags=re.DOTALL)

# Remove HTML tags
text = re.sub(r'<[^>]+>', ' ', html)

# Clean up whitespace
text = re.sub(r'\s+', ' ', text).strip()

with open('extracted.txt', 'w', encoding='utf-8') as f:
    f.write(text)

print("Extraction complete.")
